packages/util-cadence/src/index.js

```diff
@@ -0,0 +1,47 @@
+import {invariant} from "@onflow/util-invariant"
+
+/**
+ * Parses a Cadence transaction to extract function arguments.
+ *
+ * @param {string} cadenceCode - The Cadence transaction code.
+ * @returns {Array<object>} An array of objects each containing 'name' and 'type' of each argument in the transaction.
+ * Each object has the format: { name: string, type: string }. If no valid arguments are found, or if the Cadence
+ * transaction is not properly formatted, the function will return an empty array.
+ *
+ * @example
+ * const cadenceCode = 'transaction(name: String, age: Int)'
+ * parseArguments(cadenceCode)
+ * // Returns: [{ name: 'name', type: 'String' }, { name: 'age', type: 'Int' }]
+ *
+ * @example
+ * const cadenceCode = 'transaction()'
+ * parseArguments(cadenceCode)
+ * // Returns: []
+ */
+export const parseArguments = cadenceCode => {
+  const mainFunctionIndex = cadenceCode.indexOf('pub fun main')
```

Comment:
I think this should be a regex like `/pub\s+fun\s+main/` since this is valid cadence code and all valid cadence scripts should also be valid in FCL